### ISSUE TITLE
bug: TikTok: Failed to apply patches: Dependency errors with 'Settings' causing multiple failures

### ISSUE THREAD

**Issue by 54m4d** [ReVanced/revanced-patches#2748]
### Type

Error while patching

### Bug description

Getting this error while patching TikTok 30.7.2

### Steps to reproduce

Just try to patch TikTok as any other app.

### Relevant log output

```shell
Initializing installer
Creating working directory
Copying original apk
Unpacking input apk
Reading dex files
Decoding AndroidManifest.xml only, because resources are not needed
Merging integrations
An error occurred! Aborted
Error:
java.lang.OutOfMemoryError: Failed to allocate a 32 byte allocation with 4005744 free bytes and 3911KB until OOM, target footprint 536870912, growth limit 536870912; giving up on allocation because <1% of heap free after GC.
	at java.util.LinkedHashMap.newNode(LinkedHashMap.java:280)
	at java.util.HashMap.putVal(HashMap.java:641)
	at java.util.HashMap.put(HashMap.java:611)
	at brut.androlib.res.data.ResType.addResource(ResType.java:49)
	at brut.androlib.res.data.ResType.addResource(ResType.java:44)
	at brut.androlib.res.decoder.ARSCDecoder.readEntry(ARSCDecoder.java:363)
	at brut.androlib.res.decoder.ARSCDecoder.readTableType(ARSCDecoder.java:308)
	at brut.androlib.res.decoder.ARSCDecoder.readTableTypeSpec(ARSCDecoder.java:223)
	at brut.androlib.res.decoder.ARSCDecoder.readTablePackage(ARSCDecoder.java:132)
	at brut.androlib.res.decoder.ARSCDecoder.readTableHeader(ARSCDecoder.java:84)
	at brut.androlib.res.decoder.ARSCDecoder.decode(ARSCDecoder.java:50)
	at brut.androlib.res.AndrolibResources.getResPackagesFromApk(AndrolibResources.java:778)
	at brut.androlib.res.AndrolibResources.loadMainPkg(AndrolibResources.java:66)
	at brut.androlib.res.AndrolibResources.getResTable(AndrolibResources.java:58)
	at brut.androlib.Androlib.getResTable(Androlib.java:74)
	at app.revanced.patcher.Patcher.decodeResources(Patcher.kt:204)
	at app.revanced.patcher.Patcher.access$decodeResources(Patcher.kt:38)
	at app.revanced.patcher.Patcher$executePatches$1.invokeSuspend(Patcher.kt:351)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlin.sequences.SequenceBuilderIterator.hasNext(SequenceBuilder.kt:129)
	at app.revanced.manager.flutter.MainActivity.runPatcher$lambda-32(MainActivity.kt:395)
	at app.revanced.manager.flutter.MainActivity.$r8$lambda$efRWk2Z7aIFbIqnpE9hv8LpAiuk(Unknown Source:0)
	at app.revanced.manager.flutter.MainActivity$$ExternalSyntheticLambda13.run(Unknown Source:24)
	at java.lang.Thread.run(Thread.java:1012)
```

### Screenshots or videos

N/A

### Solution

N/A

### Additional context

No special context.

Invalidly closing the issue is not acceptable.

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.

**Comment by BrianPurgert** [comment]
I get the same error I've tried a couple different versions of TikTok

**Comment by 54m4d** [comment]
Manager `v1.6.1`

**Comment by 54m4d** [comment]
I've never succeeded in patching TikTok before, even with older Manager/TikTok versions, but today i decided to open the issue.

**Comment by 54m4d** [comment]
I'll try to report it @Manger, and i'll try using cli when i have the chance, Thanks!

**Comment by 54m4d** [comment]
I saw this [comment](https://github.com/ReVanced/revanced-manager/issues/647#issuecomment-1385626405) there, so prolly not a Manager issue if that person is right.

**Comment by io43** [comment]
yea it don't work, have to use the cli instead

**Comment by karx1** [comment]
Hi, I solved it by using the CLI, you can read [this Reddit thread](https://www.reddit.com/r/revancedapp/comments/11b6zy4/how_to_use_revanced_cli_for_nonroot_users_step_by/) for a guide

**Comment by 54m4d** [comment]
I'm a Manager user.

**Comment by Axelen123** [comment]
> This issue should be moved to Manager repository. This issue only shows up when patching with Manager.
> 
> Manager already has `android:largeHeap="true"` set.
> 
> But perhaps the Flutter/Dart settings can be adjusted? I've never used Flutter or Dart so I can't comment on this much. Maybe [this](https://stackoverflow.com/a/54017161) is the fix?

I don't think anything can be done. The problem is caused by the Android runtime not giving us enough java heap memory. People who run into this have to use the CLI in Termux instead, which does not have this memory limit.

**Comment by 54m4d** [comment]
1. You guys need to agree if this is a Manager issue or not, as you may have seen, oSumAtrIX mentioned that it's not related to the Manager, but at least they gave a comment before closing that issue.

2. If CLI is the only way to patch this, TikTok patches may be temporarily or permanently disabled in the Manager, with users being informed to use the CLI instead, in order to avoid confusion.

**Comment by 54m4d** [comment]
Fixed in Manager v1.7.0

**Comment by Axelen123** [comment]
> There is no such thing as "Disabling" patches. TikTok can be patched on ReVanced Manager if you have enough memory, but the issue is inherently large memory usage due to dexlib2. ReVanced Manager can fix it by increasing max memory usage.

We already did and its still not enough

**Comment by 54m4d** [comment]
I have tried the exact same process i tried with the previous Manager and i got no issues now, although TikTok version i tried with now is 30.7.4. but i doubt that is what solved it.

**Comment by Axelen123** [comment]
Termux does not run an entire operating system. It just launches linux processes. Only dalvik code is subjected to the problematic memory limit, so it does not apply when running the JRE in Termux

**Comment by 54m4d** [comment]
I believe i spoke a bit too soon. When i tried an older version of TikTok, i encountered a different error unrelated to memory issues.

The suggested version of Tiktok for patching should not be set as "All," as some versions work just fine, like `30.7.4` while others can throw errors like the one below, re-opening this issue since ReVanced Patches do not work properly with TikTok.

```Kotlin
Initializing installer
Creating working directory
Copying original apk
Unpacking input apk
Reading dex files
Decoding AndroidManifest.xml only, because resources are not needed
Merging integrations
Executing patches
Failed to apply Downloads: 'Downloads' depends on 'Settings' but the following error was raised: Failed to resolve SettingsEntryFingerprint
Failed to apply Feed filter: 'Feed filter' depends on 'Settings' but the following error was raised: 'Settings' did not succeed previously
Applied Hide ads
Failed to apply Playback speed: NullPointerException
Failed to apply Settings: 'Settings' did not succeed previously
Failed to apply Sim spoof: 'Sim spoof' depends on 'Settings' but the following error was raised: 'Settings' did not succeed previously
Repacking patched apk
Not compiling resources because resource patching is not required
Writing modified dex files
Finished!
```

**Comment by oSumAtrIX** [comment]
This is not a patch issue. There is simply not enough RAM provided to patch the app on your end.

**Comment by 54m4d** [comment]
So enough RAM is only provided to certain TikTok apks but not others, is that what you are trying to say?

In the last error, how can you tell if enough RAM was not provided?

**Comment by oSumAtrIX** [comment]
Enough RAM is needed to patch TikTok. The last error is a product of patching an old version.

**Comment by 54m4d** [comment]
Forget the memory issue now, it has been addressed in Manager v1.7.0. and i tested and got no memory errors.

The last issue is not related to memory and the suggested version of Tiktok for patching is set to"All," so i don't understand why you mention "old version"? read my posts once again, you seem like you have a hard time distinguishing between a memory issue and something else, please re-open this issue.

@Axelen123 Can you confirm if this is a Manager issue or not?

**Comment by Axelen123** [comment]
> Can you confirm if this is a Manager issue or not?

Long answer:
The memory issue is caused by the android app runtime (ART) not giving the patcher enough java heap memory. The only way to affect how much memory you are provided is to set the `android:largeHeap` attribute in the manifest, but we have already done that. The CLI is unaffected because it does not run in ART unlike manager.

Short answer: I would say this is a manager issue, but we cannot do anything about it

**Comment by 54m4d** [comment]
> > Can you confirm if this is a Manager issue or not?
> 
> Long answer: The memory issue is caused by the android app runtime (ART) not giving the patcher enough java heap memory. The only way to affect how much memory you are provided is to set the `android:largeHeap` attribute in the manifest, but we have already done that. The CLI is unaffected because it does not run in ART unlike manager.
> 
> Short answer: I would say this is a manager issue, but we cannot do anything about it.
> 
> If you are having another problem then you need to open a new issue so it can be tracked properly.

But how come the real memory issue i had was resolved in v1.7.0?
